### PR TITLE
Add sample mock dating functionality

### PR DESCRIPTION
**Pull request by omelniz**

### CODE REVIEW

File: src/components/Dating/HeaderLeft.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  count: 0,
+};
+
+const styles = () =>
+  StyleSheet.create({
+    button: {
+      paddingHorizontal: 12,
+    },
+    badge: {
+      backgroundColor: '#F3A0F0',
```
Comment (azimgd) [inline src/components/Dating/HeaderLeft.js:32]: No use of react-native-paper's theme object

File: src/components/Dating/index.js [hunk 2]
```diff
@@ -1,49 +1,26 @@
...
+const Dating = ({ items }) => {
+  const styling = styles();
 
   return (
-    <View style={styling.root}>
-      <Title style={styling.title}>{t('REAL Dating Coming Soon')}</Title>
-      <Paragraph style={styling.paragraph}>{t('REAL is fully Open Source & built by the people')}. {t('Help us move faster by contributing code')}.</Paragraph>
+    <View style={styling.content}>
+      <CardsStack items={items}>
+        {({ item, api }) => <DatingCard {...item} onAccept={api.swipeRight} onDecline={api.swipeLeft} />}
```
Comment (azimgd) [inline src/components/Dating/index.js:12]: Looks good

File: src/components/Dating/index.js [hunk 3]
```diff
@@ -1,49 +1,26 @@
...
-})
+  );
+};
 
-Dating.propTypes = {
-  theme: PropTypes.any,
-  t: PropTypes.any,
-}
+const styles = () =>
+  StyleSheet.create({
```
Comment (azimgd) [inline src/components/Dating/index.js:19]: StyleSheet create is primarily used for styling object caching. It wouldn't make much sense to use it this way as your function will be re-created each time component re-renders therefore we can omit it.

File: src/components/DatingCard/card.spec.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
```
Comment (azimgd) [inline src/components/DatingCard/card.spec.js:1]: Nice! Tests!

File: src/components/DatingMatches/Row.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import { Text, Image, View, StyleSheet } from 'react-native';
+import UserRowComponent from 'templates/UserRow';
+import DefaultButton from 'components/Formik/Button/DefaultButton';
+
+const DatingMatchesRow = ({ t, id, photo, name, age, matchedDate, onRemove, onMessage }) => {
+  const styling = styles();
+
+  const handleRemove = () => onRemove({ id });
+
+  const renderAvatar = () => (
```
Comment (azimgd) [inline src/components/DatingMatches/Row.js:14]: Good split for components, would be even better to move it out of this functions scope

File: src/store/ducks/dating/actions.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { createAction } from 'redux-actions';
+import * as constants from 'store/ducks/dating/constants';
+
+/**
+ *
+ */
+export const datingSearchRequest = createAction(constants.DATING_SEARCH_REQUEST);
+export const datingSearchSuccess = createAction(constants.DATING_SEARCH_SUCCESS);
```
Comment (azimgd) [inline src/store/ducks/dating/actions.js:8]: Missing Failure and Idle actions

File: src/store/ducks/dating/actions.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ *
+ */
+export const datingMatchedRequest = createAction(constants.DATING_MATCHED_REQUEST);
+export const datingMatchedSuccess = createAction(constants.DATING_MATCHED_SUCCESS);
+
+/**
+ *
+ */
+export const datingMatchedRemove = createAction(constants.DATING_MATCHED_REMOVE);
```
Comment (azimgd) [inline src/store/ducks/dating/actions.js:19]: Invalid action notation, we usually want our actions to end with request/success etc.

File: src/store/ducks/dating/dating.spec.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { combineReducers } from "redux";
```
Comment (azimgd) [inline src/store/ducks/dating/dating.spec.js:1]: nice!

File: src/store/ducks/dating/reducer.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  update(state, {
+    datingMatched: {
+      data: { $set: action.payload.data },
+    },
+  });
+
+const datingMatchedRemove = (state, action) =>
+  update(state, {
+    datingMatched: {
+      data: { $set: state.datingMatched.data.filter(({ id }) => id !== action.payload.id) },
```
Comment (azimgd) [inline src/store/ducks/dating/reducer.js:37]: It's ok to modify data this way, but usually we either modify it before saving into reducers or use `extend` method from `immutability-helpers` to simplify our reducers

File: src/store/ducks/dating/saga.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    photo: path(['picture', 'large'], item),
+    name: path(['name', 'first'], item),
+    age: path(['dob', 'age'], item),
+    city: path(['location', 'city'], item),
+  }));
+}
+
+function* datingSearchRequest() {
+  try {
+    const response = yield fetch('https://randomuser.me/api/?page=1&results=100&seed=abc');
```
Comment (azimgd) [inline src/store/ducks/dating/saga.js:20]: This is alright, but would be better to have extra function called `handleDatingSearchRequest` which will contain an adapter for this request.

File: src/templates/CardsStack/index.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
+import React, { PureComponent } from "react";
+import PropTypes from "prop-types";
+import { StyleSheet } from "react-native";
+import CardStack, { Card } from "react-native-card-stack-swiper";
+
+class CardsStack extends PureComponent {
```
Comment (azimgd) [inline src/templates/CardsStack/index.js:6]: We don't use react classes in this project (just at couple places only when they are unavoidable)

File: src/templates/CardsStack/index.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const { items, onSwipedAll, onSwipedLeft, onSwipedRight, children } = this.props;
```
Comment (azimgd) [inline src/templates/CardsStack/index.js:12]: no need for spreading props

File: .prettierrc.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  semi: true,
+  trailingComma: "all",
+  singleQuote: true,
+  printWidth: 120,
+  tabWidth: 2,
+};
```
Comment (mfogel) [inline .prettierrc.js:7]: I like prettier and use it as well, but in the interest of keeping PR's focused on idea at a time, I would have left this out.

File: src/components/DatingCard/card.spec.js (hunk 4 above)
Comment (mfogel) [reply src/components/DatingCard/card.spec.js:1]: Yay tests!

File: src/components/DatingMatches/header.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import { useEffect } from 'react';
+import { pageHeaderLeft } from 'navigation/options';
+
+export const useHeader = ({ goBack, navigation }) => {
+  const headerLeft = (props) => pageHeaderLeft({ ...props, onPress: goBack });
+
+  /**
+   *
+   */
```
Comment (mfogel) [inline src/components/DatingMatches/header.js:9]: I would either fill in the comment or remove the placeholder

**Pull request closed without merging**